### PR TITLE
Support "noUpdate" signal in Feeds

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/38056
https://github.com/demisto/etc/issues/24110

## Description
Added the method `get_no_update_value` to the scripts `HTTPFeedApiModule` and `JSONFeedApiModule`,
This method reads the `last_modified` and `etag` headers from the feed response and save it to the integration context, when the feed fetch indicators again the method gets the values from the headers again and compare it with the values from the context, if the values for `last_modified` or `etag` changed it means that the feed response updated and we need to call the command `demisto.createIndicators` with `noUpdate=False`

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FeedIntel471/ReleaseNotes/2_0_2.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+#### Integrations
+
+##### Intel471 Actors Feed
+- Read ETAG and Last-Modified headers from feed response to supports `noUpdate` signaling when creating indicators.
+
+##### Intel471 Malware Feed (Deprecated)
```
Comment (yaakovi) [inline Packs/FeedIntel471/ReleaseNotes/2_0_2.md:7]: revert this one, no need to update a deprecated feed.

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 2]
```diff
@@ -280,6 +280,37 @@ def custom_fields_creator(self, attributes: dict):
...
+    https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified
+    https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag
+    Args:
+        response: (requests.Response) The feed response.
+    Returns:
+        boolean with the value for noUpdate argument.
+        The value should be False if the response was modified.
+    """
+
+    context = demisto.getIntegrationContext()
```
Comment (yaakovi) [inline Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:296]:
```suggestion
    context = get_integration_context()
```

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 3]
```diff
@@ -280,6 +280,37 @@ def custom_fields_creator(self, attributes: dict):
...
+    old_etag = context.get('etag')
+    old_last_modified = context.get('last_modified')
+
+    etag = response.headers.get('ETag')
+    last_modified = response.headers.get('Last-Modified')
+
+    demisto.setIntegrationContext({'last_modified': last_modified, 'etag': etag})
```
Comment (yaakovi) [inline Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:303]:
```suggestion
    set_integration_context({'last_modified': last_modified, 'etag': etag})
```

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 4]
```diff
@@ -280,6 +280,37 @@ def custom_fields_creator(self, attributes: dict):
...
+
+    if old_etag and old_etag != etag:
+        return False
+
+    if old_last_modified and old_last_modified != last_modified:
+        return False
+
+    return True
```
Comment (yaakovi) [inline Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:311]: add a debug log before each return to indicate the result (and reason)

**Review by yaakovi — COMMENTED** [review]
nice work.
the same comments apply to the JSONFeedApiModule.

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 5]
```diff
@@ -293,21 +293,26 @@ def get_no_update_value(response: requests.Response) -> bool:
...
     old_last_modified = context.get('last_modified')
 
     etag = response.headers.get('ETag')
     last_modified = response.headers.get('Last-Modified')
 
-    demisto.setIntegrationContext({'last_modified': last_modified, 'etag': etag})
+    set_integration_context({'last_modified': last_modified, 'etag': etag})
 
     if old_etag and old_etag != etag:
+        demisto.log('New indicators fetched - the ETag value has been updated,'
```
Comment (yaakovi) [inline Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:306]: switch all logs to debug

**Review by yaakovi — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@yaakovi RNs reviewed.

**Comment by ShahafBenYakir** [comment]
@adi88d @yaakovi Why for intel 471 have we bumped the version and created RN but did not make any real change to the pack?

**Comment by adi88d** [comment]
> @adi88d @yaakovi Why for intel 471 have we bumped the version and created RN but did not make any real change to the pack?

intel 471 based on `JSONFeedApiModule` which changed

**Comment by ShahafBenYakir** [comment]
Thanks!

**Pull request merged**